server/src/main/java/org/opensearch/index/translog/NoOpTranslogManager.java

```diff
@@ -112,4 +112,10 @@ public Translog.Operation readOperation(Translog.Location location) throws IOExc
     public Translog.Location add(Translog.Operation operation) throws IOException {
         return new Translog.Location(0, 0, 0);
     }
+
+    @Override
+    public Translog.Snapshot newChangesSnapshot(long fromSeqNo, long toSeqNo, boolean requiredFullRange) throws IOException,
+        UnsupportedOperationException {
+        throw new UnsupportedOperationException("Translog snapshot unsupported with no-op translogs");
+    }
 }
```

Comment:
I would suggest we create and return Translog.Snapshot.EMPTY_SNAPSHOT
```
Translog.Snapshot EMPTY_SNAPSHOT = new Translog.Snapshot() {
            @Override
            public void close() {}

            @Override
            public int totalOperations() {
                return 0;
            }

            @Override
            public Translog.Operation next() {
                return null;
            }
        };
```

Reply:
Empty snapshot can imply that there are no operations to be replayed which can be a valid scenario for us. 

What we need is a way for the caller to know that current `TranslogManager` doesn't support this use case. How about we throw an exception(UnsupportedOperationException or RunTimeException) for noop cases and let the [caller](https://github.com/opensearch-project/cross-cluster-replication/blob/4f790d20eeefe78f00c0810501e73bc2b5dfc1ef/src/main/kotlin/org/opensearch/replication/action/changes/TransportGetChangesAction.kt#L103-L125) fallback to lucene for such cases. 
I'll also remove the `UnsupportedOperationException`  from the interface like we discussed.

Reply:
I think that's a reasonable approach.